### PR TITLE
Coordinate searsh fix

### PR DESCRIPTION
**Pull request by nnngrach**
[102](https://github.com/osmandapp/OsmAnd-Issues/issues/102)

[video](https://www.dropbox.com/s/rczmg61sinxed0w/Video%2019.10.2021%2C%2010%2011%2022.mp4?dl=0)
[video2](https://www.dropbox.com/s/j1vk2bd5m3x4ibx/Screen%20Recording%202021-10-21%20at%2010.04.31.mov?dl=0)

### CODE REVIEW

File: Sources/Controllers/TargetMenu/Search/OAQuickSearchCoordinateFormatsViewController.m [hunk 1]
```diff
@@ -48,13 +50,8 @@ - (void) commonInit
...
 {
     _data = [NSMutableArray array];
-    
-    OAMapPanelViewController *mapPanel = [OARootViewController instance].mapPanel;
-    CLLocation *location = [OsmAndApp instance].locationServices.lastKnownLocation;
-    if (!location)
-        location = mapPanel.mapViewController.getMapLocation;
-    double lat = location.coordinate.latitude;
-    double lon = location.coordinate.longitude;
+    double lat = _location.coordinate.latitude;
```
Comment (PaulStets) [inline Sources/Controllers/TargetMenu/Search/OAQuickSearchCoordinateFormatsViewController.m:53]: Maybe you should still check for nil?

File: Sources/Controllers/TargetMenu/Search/OAQuickSearchCoordinatesViewController.mm [hunk 2]
```diff
@@ -944,16 +957,19 @@ - (BOOL)textFieldShouldReturn:(UITextField *)textField {
...
-    CGFloat duration = [[userInfo objectForKey:UIKeyboardAnimationDurationUserInfoKey] floatValue];
-    NSInteger animationCurve = [[userInfo objectForKey:UIKeyboardAnimationCurveUserInfoKey] integerValue];
-    NSValue *keyboardBoundsValue = [userInfo objectForKey:UIKeyboardFrameEndUserInfoKey];
-    CGFloat keyboardHeight = [keyboardBoundsValue CGRectValue].size.height;
-    
-    [UIView animateWithDuration:duration delay:0. options:animationCurve animations:^{
-        UIEdgeInsets insets = [[self tableView] contentInset];
-        [[self tableView] setContentInset:UIEdgeInsetsMake(insets.top, insets.left, kHintBarHeight + (DeviceScreenHeight - keyboardHeight), insets.right)];
-    } completion:nil];
+    if ([OAUtilities isLandscape])
```
Comment (PaulStets) [inline Sources/Controllers/TargetMenu/Search/OAQuickSearchCoordinatesViewController.mm:960]: Only react to keyboard in landscape?

File: Sources/OAWorldRegion.mm [hunk 3]
```diff
@@ -621,16 +621,9 @@ - (OAWorldRegion *) findAtLat:(double)latitude lon:(double)longitude
...
-    for (OAWorldRegion *region in list)
-    {
-        if ([region contain:latitude lon:longitude])
-        {
-            NSString *name = region.name;
-            if (name && name.length > 0)
-                return name;
-        }
-    }
+    OAWorldRegion *region = [self findAtLat:latitude lon:longitude];
```
Comment (PaulStets) [inline Sources/OAWorldRegion.mm:624]: Is this correct? It changes the logic of the method greatly. Will this break other places where this code is used?
/

File: Sources/Controllers/TargetMenu/Search/OAQuickSearchCoordinateFormatsViewController.m (hunk 1 above)
Comment (nnngrach) [reply Sources/Controllers/TargetMenu/Search/OAQuickSearchCoordinateFormatsViewController.m:58]: Added a check for nil

**Pull request merged**